File: ant-sdk/antd-php/examples/03-chunks.php
<?php

/**
 * Example 03: Store and retrieve raw chunks.
 *
 * Prerequisites: antd daemon running with a funded wallet.
 */

require_once __DIR__ . '/../vendor/autoload.php';

use Autonomi\Antd\AntdClient;

$client = new AntdClient();

// Store a chunk
$result = $client->chunkPut('raw chunk data');
echo "Chunk stored at: {$result->address} (cost: {$result->cost} atto)\n";

// Retrieve the chunk
$data = $client->chunkGet($result->address);
echo "Retrieved chunk: {$data}\n";

File: ant-sdk/antd-rust/examples/03-chunks.rs
use antd_client::{Client, DEFAULT_BASE_URL};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::new(DEFAULT_BASE_URL);

    // Store a raw chunk
    let result = client.chunk_put(b"raw chunk data").await?;
    println!(
        "Chunk stored at: {} (cost: {} atto)",
        result.address, result.cost
    );

    // Retrieve the chunk
    let data = client.chunk_get(&result.address).await?;
    println!("Chunk data: {} bytes", data.len());

    Ok(())
}
